### DISCUSSION TITLE
Unable to use constraints files in a simple minimization procedure

### DISCUSSION THREAD

**Discussion by oroscow** [RosettaCommons/rosetta#253] [Rosetta] [answered]
I'm trying to replicate [this example](https://docs.rosettacommons.org/demos/latest/tutorials/minimization/minimization#how-to-minimization-with-constraints). Here's a minimal code example:

**Directory:**
```
- 3hon.pdb
- 3hon.pdb_0001.pdb
- cstfile.cst
- min_flag
```

**Flag file:**
```
-s 3hon.pdb
-run:min_tolerance 0.001
-constraints:cst_file cstfile
-score:weights ref2015_cst
-out:suffix _minwithcsts
```

**Constraints file:**
```AtomPair OE2 165 OH 305 CONSTANTFUNC 2
AtomPair CA 111 CA 249 CONSTANTFUNC 5
Angle OE2 363 NH2 76 CZ 76 CONSTANTFUNC 6
Dihedral CG 221 ND2 221 OE1 165 CD 165 CONSTANTFUNC 7
AtomPair CA 445 OH 445 CONSTANTFUNC 9
Dihedral CG 25 ND2 25 CA 25 N 25  CONSTANTFUNC 8
```
I got the contents of my constraints file [here](https://github.com/RosettaCommons/rosetta/blob/4e95d564dc8b93fbe830018d2f295f3f69a70dfd/tests/integration/tests/residue_energy_breakdown/input/cstfile.cst#L4).

**Rosetta command:**
```
minimize @min_flag
```

**Crash log (abbreviated):**
```
[FILE]: src/core/scoring/constraints/ConstraintIO.cc
[LINE]: 454
[START_MESSAGE]
[ ERROR ] UtilityExitException
ERROR: [ERROR] Unable to open constraints file: cstfile
```

**Comment by roccomoretti** [comment, marked as answer]
Rosetta doesn't (necessarily) add the .cst extension to constraint files. You'll need to use `-constraints:cst_file cstfile.cst` for Rosetta to properly recognize the file `cstfile.cst` in the working directory.

**Reply by oroscow** [reply]
As usual, a simple solution; thanks so much!
